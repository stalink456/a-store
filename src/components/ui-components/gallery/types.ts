export type GalleryType = {
  images: string[];
};
