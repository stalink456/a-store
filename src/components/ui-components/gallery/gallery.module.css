@import "../../../css/variables.css";

.gallery {
  padding: 0 15px;
  width: 600px;
}

.gallery img {
  width: 100%;
  height: 600px;
  object-fit: cover;
  object-position: top 20%;
}

.gallery .gallery__small_pictures {
  max-width: 100px;
}
.gallery_container img {
  border: 2px solid var(--grey-color);
}

.gallery__small_pictures img {
  height: 60px;
  width: 60px;
}

@media (max-width: 1165px) {
  .gallery {
    margin-bottom: 20px;
  }
}

@media (max-width: 1065px) {
  .gallery {
    width: 640px;
  }
}

@media (max-width: 660px) {
  .gallery {
    margin-right: 15px;
    width: 100%;
  }
}
