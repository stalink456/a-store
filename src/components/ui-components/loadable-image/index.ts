export { LoadableImage } from './loadable-image';
