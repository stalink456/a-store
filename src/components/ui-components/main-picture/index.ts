export { MainPicture } from "./main-picture";
