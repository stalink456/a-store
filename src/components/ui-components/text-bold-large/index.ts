export { TextBoldLarge } from "./text-bold-large";
