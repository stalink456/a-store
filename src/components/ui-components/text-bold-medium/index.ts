export { TextBoldMedium } from "./text-bold-medium";
