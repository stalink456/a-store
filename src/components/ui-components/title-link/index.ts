export { TitleLink } from './title-link';
