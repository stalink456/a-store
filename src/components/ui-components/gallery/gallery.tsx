import React from 'react';
import { Space } from '@alfalab/core-components/space';
import { LoadableImage } from 'components/ui-components/loadable-image';
import { GalleryType } from './types';
import { useGallery } from 'hooks/use-gallery';

import styles from './gallery.module.css';

export const Gallery: React.FC<GalleryType> = ({ images }) => {
  const { sliderImage, handleOnClick } = useGallery(images);

  return (
    <div className={styles.gallery}>
      <Space direction='vertical' fullWidth>
        <LoadableImage key={sliderImage} src={sliderImage} alt={sliderImage} />
        <Space direction='horizontal' size={5} wrap>
          {images.map((value, id) => (
            <div
              className={value === sliderImage ? styles.gallery_container : ''}
              key={id}
              onClick={() => handleOnClick(value)}
            >
              <Space
                direction='horizontal'
                className={styles.gallery__small_pictures}
              >
                <LoadableImage src={value} alt={value} />
              </Space>
            </div>
          ))}
        </Space>
      </Space>
    </div>
  );
};
