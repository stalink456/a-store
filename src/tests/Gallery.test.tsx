import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Gallery } from 'components/ui-components/gallery';
import * as images from './mocks/get-images-200.json';

const mockedImages = images['images'];
const firstImage = mockedImages[0];
const secondImage = mockedImages[1];

describe('Gallery component', () => {
  it('should render Gallery component with images', () => {
    const { baseElement } = render(<Gallery images={mockedImages} />, {
      wrapper: MemoryRouter,
    });

    expect(baseElement).toMatchSnapshot();
  });

  it('should render Gallery component with two images', async () => {
    render(<Gallery images={[firstImage]} />, { wrapper: MemoryRouter });

    expect(await screen.findAllByRole('img')).toHaveLength(2);
  });

  it('should render Gallery component with switching images', async () => {
    render(<Gallery images={mockedImages} />, { wrapper: MemoryRouter });

    expect(await screen.findAllByAltText(firstImage))[0] as HTMLImageElement;

    fireEvent.click(screen.getByAltText(secondImage));

    expect(await screen.findAllByAltText(secondImage))[0] as HTMLImageElement;
  });

  it("shouldn't render Gallery component with non-existent image", async () => {
    render(<Gallery images={mockedImages} />, { wrapper: MemoryRouter });

    const image = screen.queryByAltText(
      'non-existent image'
    ) as HTMLImageElement;
    await waitFor(() => expect(image === null).toBeTruthy());
  });

  it("shouldn't render Gallery component with non-existent description", async () => {
    render(<Gallery images={mockedImages} />, { wrapper: MemoryRouter });

    const result = screen.queryByAltText(
      'non-existent description'
    ) as HTMLImageElement;
    await waitFor(() => expect(result).not.toBeTruthy());
  });
});
