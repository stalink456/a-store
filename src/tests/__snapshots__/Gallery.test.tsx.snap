// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should render Gallery component with images 1`] = `
<body>
  <div>
    <div
      class="gallery"
    >
      <div
        class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          style="margin-bottom: 16px;"
        >
          <div
            class="skeleton__component_1bshy skeleton__animate_1bshy"
          >
            <img
              alt="http://qa-games.ru/astore/public/images/61646585.png"
              class=""
              src="http://qa-games.ru/astore/public/images/61646585.png"
            />
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <div
            class="space__spaceContainer_nr3as space__start_nr3as"
            style="flex-wrap: wrap; margin-bottom: -5px;"
          >
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class="gallery_container"
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/61646585.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/61646585.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/29918301.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/29918301.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/23597101.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/23597101.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/79160052.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/79160052.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/34570797.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/34570797.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/95291281.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/95291281.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/1_3d.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/1_3d.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/2_3d.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/2_3d.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="margin-right: 5px; padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/45157942.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/45157942.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as"
              style="padding-bottom: 5px;"
            >
              <div
                class=""
              >
                <div
                  class="space__spaceContainer_nr3as space__start_nr3as gallery__small_pictures"
                >
                  <div
                    class="space__spaceItem_nr3as"
                  >
                    <div
                      class="skeleton__component_1bshy skeleton__animate_1bshy"
                    >
                      <img
                        alt="http://qa-games.ru/astore/public/images/Frame_117.png"
                        class=""
                        src="http://qa-games.ru/astore/public/images/Frame_117.png"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
